### PR TITLE
add load test environement and fixtures

### PR DESCRIPTION
**Pull request by itkg-rboessel**

### CODE REVIEW

**Comment by itkg-rboessel** [comment]
https://github.com/open-orchestra/open-orchestra-front-demo/pull/73

File: app/AppKernel.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ public function registerBundles()
             new AntiMattr\Bundle\MongoDBMigrationsBundle\MongoDBMigrationsBundle(),
         );
 
-        if (in_array($this->getEnvironment(), array('dev', 'test'))) {
+        if (in_array($this->getEnvironment(), array('dev', 'test', 'load_test'))) {
```
Comment (itkg-alavieille) [inline app/AppKernel.php]: i think the debugBundle, WebProfiler and SensioDistribution are not useful for the load_test environnement

File: web/app_load_test.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+use Symfony\Component\ClassLoader\ApcClassLoader;
+use Symfony\Component\HttpFoundation\Request;
+
+$loader = require_once __DIR__.'/../app/bootstrap.php.cache';
+
+// Use APC for autoloading to improve performance.
```
Comment (itkg-alavieille) [inline web/app_load_test.php]: deleted the comment

File: jmeter/data/node.csv [hunk 3]
```diff
@@ -1,5 +1,50 @@
-root,/
```
Comment (itkg-alavieille) [inline jmeter/data/node.csv]: is it possible to replace the csv file by a loop in jmeter ?

**Comment by itkg-alavieille** [comment]
i don't understand your fixture why duplicate the fixture  group, site ...

File: src/DataFixtures/MongoDB/LoadGroupData.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace OpenOrchestra\LoadTestBundle\MongoDB\DataFixtures;;
```
Comment (nicolasThal) [inline src/DataFixtures/MongoDB/LoadGroupData.php]: the namespace does not match the folder

**Comment by itkg-rboessel** [comment]
@itkg-alavieille @nicolasThal @itkg-ngilain @itkg-nanne

File: jmeter/routing.jmx [hunk 5]
```diff
@@ -525,6 +523,15 @@
         <hashTree/>
         <RecordingController guiclass="RecordController" testclass="RecordingController" testname="Recording Controller" enabled="true"/>
         <hashTree>
+          <CounterConfig guiclass="CounterConfigGui" testclass="CounterConfig" testname="Compteur" enabled="true">
+            <stringProp name="CounterConfig.start">0</stringProp>
+            <stringProp name="CounterConfig.end">99</stringProp>
```
Comment (itkg-alavieille) [inline jmeter/routing.jmx]: can you convert this parameter in variable for command.  like this for instance : ${__P(nbr_node,99)}

File: jmeter/routing.jmx [hunk 6]
```diff
@@ -159,6 +148,15 @@
               <boolProp name="TransactionController.parent">false</boolProp>
             </TransactionController>
             <hashTree>
+              <CounterConfig guiclass="CounterConfigGui" testclass="CounterConfig" testname="Compteur" enabled="true">
+                <stringProp name="CounterConfig.start">0</stringProp>
+                <stringProp name="CounterConfig.end">99</stringProp>
```
Comment (itkg-alavieille) [inline jmeter/routing.jmx]: can you convert this parameter in variable for command.  like this for instance : ${__P(nbr_node,99)}

File: src/DataFixtures/MongoDB/LoadNodeData.php [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+            $this->generateSimpleNode($name, 'en', $manager, $content, $pattern);
+            $this->generateSimpleNode($name, 'fr', $manager, $content, $pattern);
+        }
+        $manager->flush();
+    }
+
+    /**
+     * @return int
+     */
+    public function getOrder()
```
Comment (itkg-alavieille) [inline src/DataFixtures/MongoDB/LoadNodeData.php]: this method isn't used

File: src/DataFixtures/MongoDB/LoadNodeData.php [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+        $node->setVersion(1);
+        $node->setLanguage($language);
+        $node->setDeleted(false);
+        $node->setCreatedAt(new \Datetime());
+        $node->setUpdatedAt(new \Datetime());
+        $node->setStatus($value);
+
+        return $node;
+    }
+
```
Comment (itkg-alavieille) [inline src/DataFixtures/MongoDB/LoadNodeData.php]: remove this line

**Comment by itkg-alavieille** [comment]
May be move the fixtures in other folder than MongoDB (MongoDBLoadTest for instance) thus the fixture for load test do not load when execute commnad ``app/console  doctrine:mongodb:fixtures:load`` and before load test execute this command ``app/console  doctrine:mongodb:fixtures:load --fixtures=/path/to/fixture --append`` .

Can you update the task capistrano for load fixture in the load_test environment before run test

**Comment by itkg-rboessel** [comment]
@itkg-alavieille fixtures are not in a bundle so its need to call --fixtures=src/ option to be call. And i think jmeter test will be launch occasionnaly, so i'm not sur it's useful to load them each time.

**Comment by itkg-rboessel** [comment]
@itkg-alavieille @nicolasThal @itkg-nanne

File: web/app_load_test.php [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+// Change 'sf2' to a unique prefix in order to prevent cache key conflicts
+// with other applications also using APC.
+$apcLoader = new ApcClassLoader('open_orchestra', $loader);
+$loader->unregister();
+$apcLoader->register(true);
+
+require_once __DIR__.'/../app/AppKernel.php';
+//require_once __DIR__.'/../app/AppCache.php';
```
Comment (itkg-alavieille) [inline web/app_load_test.php]: remove this comment line

File: web/app_load_test.php [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+$kernel = new AppKernel('load_test', false);
+$kernel->loadClassCache();
+//$kernel = new AppCache($kernel);
```
Comment (itkg-alavieille) [inline web/app_load_test.php]: remove this comment line

File: web/app_load_test.php [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+// When using the HttpCache, you need to call the method in your front controller instead of relying on the configuration parameter
+//Request::enableHttpMethodParameterOverride();
```
Comment (itkg-alavieille) [inline web/app_load_test.php]: remove this comment line

**Comment by itkg-alavieille** [comment]
:+1: when you are remove jmeter.log and add it in gitignore

**Pull request merged**